src/Compilers/CSharp/Test/Emit/CodeGen/SwitchTests.cs

```diff
@@ -9718,5 +9718,1207 @@ .locals init (int V_0)
+
+    }
+    static int M(uint d)
+    {
+        return d switch
+        {
+            >= 27U and < 29U => 19,
+            26U => 18,
+            9U => 5,
+            >= 2U and < 4U => 1,
+            12U => 8,
+            >= 21U and < 23U => 15,
+            19U => 13,
+            29U => 20,
+            >= 13U and < 15U => 9,
+            10U => 6,
+            15U => 10,
+            11U => 7,
+            4U => 2,
+            >= 16U and < 18U => 11,
+            >= 23U and < 25U => 16,
+            18U => 12,
+            >= 7U and < 9U => 4,
+            25U => 17,
+            20U => 14,
+            >= 5U and < 7U => 3,
+            _ => 0,
+        };
+    }
+}
+";
+            var expectedOutput =
+@"1
+1
+2
+3
+3
+4
+4
+5
+6
+7
+8
+9
+9
+10
+11
+11
+12
+13
+14
+15
+15
+16
+16
+17
+18
+19
+19
+20
+";
+            var compVerifier = CompileAndVerify(source,
+                options: TestOptions.ReleaseExe.WithOutputKind(OutputKind.ConsoleApplication),
+                parseOptions: TestOptions.RegularWithPatternCombinators,
+                expectedOutput: expectedOutput);
+            compVerifier.VerifyIL("C.M", @"
+    {
+      // Code size      243 (0xf3)
+      .maxstack  2
+      .locals init (int V_0)
+      IL_0000:  ldarg.0
+      IL_0001:  ldc.i4.s   21
+      IL_0003:  blt.un.s   IL_0039
+      IL_0005:  ldarg.0
+      IL_0006:  ldc.i4.s   27
+      IL_0008:  blt.un.s   IL_001f
+      IL_000a:  ldarg.0
+      IL_000b:  ldc.i4.s   29
+      IL_000d:  blt.un     IL_0093
+      IL_0012:  ldarg.0
+      IL_0013:  ldc.i4.s   29
+      IL_0015:  beq        IL_00b3
+      IL_001a:  br         IL_00ef
+      IL_001f:  ldarg.0
+      IL_0020:  ldc.i4.s   23
+      IL_0022:  blt.un     IL_00a9
+      IL_0027:  ldarg.0
+      IL_0028:  ldc.i4.s   25
+      IL_002a:  blt.un     IL_00d3
+      IL_002f:  ldarg.0
+      IL_0030:  ldc.i4.s   26
+      IL_0032:  beq.s      IL_0098
+      IL_0034:  br         IL_00e1
+      IL_0039:  ldarg.0
+      IL_003a:  ldc.i4.s   13
+      IL_003c:  blt.un.s   IL_0061
+      IL_003e:  ldarg.0
+      IL_003f:  ldc.i4.s   15
+      IL_0041:  blt.un.s   IL_00b8
+      IL_0043:  ldarg.0
+      IL_0044:  ldc.i4.s   18
+      IL_0046:  bge.un.s   IL_004f
+      IL_0048:  ldarg.0
+      IL_0049:  ldc.i4.s   15
+      IL_004b:  beq.s      IL_00c1
+      IL_004d:  br.s       IL_00ce
+      IL_004f:  ldarg.0
+      IL_0050:  ldc.i4.s   18
+      IL_0052:  beq        IL_00d8
+      IL_0057:  ldarg.0
+      IL_0058:  ldc.i4.s   19
+      IL_005a:  beq.s      IL_00ae
+      IL_005c:  br         IL_00e6
+      IL_0061:  ldarg.0
+      IL_0062:  ldc.i4.4
+      IL_0063:  bge.un.s   IL_006e
+      IL_0065:  ldarg.0
+      IL_0066:  ldc.i4.2
+      IL_0067:  bge.un.s   IL_00a1
+      IL_0069:  br         IL_00ef
+      IL_006e:  ldarg.0
+      IL_006f:  ldc.i4.7
+      IL_0070:  blt.un.s   IL_008d
+      IL_0072:  ldarg.0
+      IL_0073:  ldc.i4.s   9
+      IL_0075:  sub
+      IL_0076:  switch    (
```

Comment:
I didn't understand why we're producing a `switch` here for 9U, 10U, 11U and 12U, but we're not for 18U, 19U, 20U. Is there a minimum cap at 4 values?

Reply:
This test was generated automatically by `PatternMatchingTests3.RelationalFuzz_01` and is used as an example of the quality of code generated for mixed relational and constant patterns.

---
In reply to: [400475069](https://github.com/dotnet/roslyn/pull/42313#discussion_r400475069) [](ancestors = 400475069)

Reply:
Oh, I understand what you were asking.  Yes, the code to emit switches uses direct tests for 3 values (and has for years)

---
In reply to: [400567926](https://github.com/dotnet/roslyn/pull/42313#discussion_r400567926) [](ancestors = 400567926,400475069)